### PR TITLE
move guides into versioned documentation

### LINKED ISSUE #4396: Move non versioned Guide docs to versioned docs
**What steps did you take and what happened:**
Some of the Guides in the doc sit outside the versioned docs, also some are maybe not applicable to the latest version/s of Contour.
Remove older guides
Move guides under versioned docs

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Contour version:
- Kubernetes version: (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by skriss**
Closes #4396.

Couple notes:
- The versioned guides exist in "main" and "v1.23.2" docs only
- I updated the top-level site header nav: Guides is no longer listed there, and Documentation moved left to be more prominent. Guides are now in the documentation TOC.
- I left the existing non-versioned guides in place to avoid breaking old links, but I updated as many links as I could find (i.e. from the versioned docs) to point to the new versioned guides
- I deleted a few guides that were no longer relevant

Still clicking around to make sure I didn't break anything.

Preview site: https://deploy-preview-4977--projectcontour.netlify.app/

### CODE REVIEW

**Comment by skriss** [comment]
One thing I don't like is how long the nav/TOC is getting for docs, it just makes it hard to navigate. Would be nice to have the sections within the TOC be collapsible. ref. https://github.com/projectcontour/contour/issues/3815

**Comment by skriss** [comment]
I think this is looking pretty good, marked ready for review.

File: site/data/docs/main-toc.yml [hunk 1]
```diff
@@ -66,21 +66,35 @@ toc:
   - title: Guides
     subfolderitems:
       - page: AWS with NLB
-        link: /guides/deploy-aws-nlb/
+        url: /guides/deploy-aws-nlb/
```
Comment (skriss) [inline site/data/docs/main-toc.yml:69]: TODO finalize titles

**Review by sunjayBhatia — APPROVED** [review]
the deploy preview looks great 👍🏽 

looked through the commits as well, seems like you caught all the cross-page links

might be a good time to reenable or at least rerun the link checker for this change

File: site/data/docs/main-toc.yml (hunk 1 above)
Comment (skriss) [reply site/data/docs/main-toc.yml:69]: Updated these to mostly match the individual page titles, they're now a bit longer and they wrap, but I think it's OK overall.

**Comment by skriss** [comment]
> might be a good time to reenable or at least rerun the link checker for this change

Yeah didn't that break with the migration to Hugo? Will have to look and see if there's something we can use, agreed it would be a nice check to have.

**Comment by sunjayBhatia** [comment]
> > might be a good time to reenable or at least rerun the link checker for this change
> 
> Yeah didn't that break with the migration to Hugo? Will have to look and see if there's something we can use, agreed it would be a nice check to have.

yep i think that was around the time it was disabled

**Comment by sunjayBhatia** [comment]
> > might be a good time to reenable or at least rerun the link checker for this change
> 
> Yeah didn't that break with the migration to Hugo? Will have to look and see if there's something we can use, agreed it would be a nice check to have.

I'm cool with merging and doing another link check on the side and fixing things in another PR!

**Comment by skriss** [comment]
Sounds good, will merge and then find a link checker to use.

**Pull request merged**